src/models/donation.rb

```diff
@@ -1,8 +1,11 @@
 class Donation < ActiveRecord::Base
   validates_presence_of :shop, :order_id, :donation_amount
   validates_uniqueness_of :order_id, scope: :shop, conditions: -> { where("status != 'void' or status is null") }
+  validates_uniqueness_of :donation_number, scope: :shop
```

Comment:
I think this validation is problematic until the backfill is done. It might prevent updates from working